src/Gnuplot.jl

```diff
@@ -1327,6 +1327,41 @@ save(sid::Symbol; kw...) = execall(getsession(sid); kw...)
 save(             file::AbstractString; kw...) = savescript(getsession()   , file, kw...)
 save(sid::Symbol, file::AbstractString; kw...) = savescript(getsession(sid), file, kw...)
 
+# ╭───────────────────────────────────────────────────────────────────╮
+# │                     Interfacing Julia's show                      │
+# ╰───────────────────────────────────────────────────────────────────╯
+# --------------------------------------------------------------------
+
+# Define a display that will be used when Gnuplot.jl is used
+# in the Julia REPL (see PGFPlotsX.jl).
+struct GnuplotDisplay <: AbstractDisplay end
+function __init__()
+    pushdisplay(GnuplotDisplay())
+    atreplinit(i -> begin
+        if PlotDisplay() in Base.Multimedia.displays
+            popdisplay(GnuplotDisplay())
+        end
+        pushdisplay(GnuplotDisplay())
+    end)
+end
+function Base.display(d::GnuplotDisplay, gp::Session)
+    execall(gp)
+    return
+end
+function Base.show(io::IO, ::MIME"image/svg+xml", gp::Session)
+    tmpfile = tempname()*".svg"
+    execall(gp; output=tmpfile, term="svg")
+    write(io, read(tmpfile))
```

Comment:
One thing to look out for (it bit me on Gaston.jl): assuming `execall()` offloads to gnuplot and returns, `read()` may attempt to read the file before gnuplot is done (especially for a large or complex plot). This may result in incomplete data being written to `io`.

Reply:
I agree this might be an issue.
We need a way to check that gnuplot actually finished writing the output file.

A viable approach is to use a named pipe (as with `mkfifo`) in place of a regular file.  But I don't know if this works on MS Windows...

Reply:
FWIW, What I did in Gaston is to have gnuplot output a message on stderr as the last thing it does. Since gnuplot executes commands serially, I can be sure that by the time the message is seen on stderr, the plot is done.

Reply:
Thank you mbaz!
Then we may proceed by ignoring this issue.  Once the PR is ready I will add the code to ensure synchronization.